src/Features/CSharp/Portable/CodeRefactorings/CSharpRefactoringHelpersService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.ExtractMethod;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings
+{
+    [ExportLanguageService(typeof(ISyntaxTriviaService), LanguageNames.CSharp), Shared]
+    class CSharpRefactoringHelpersService : RefactoringHelpersService
+    {
+        public SyntaxNode ExtractNodeFromDeclarationAndAssignment<TNode>(SyntaxNode current) where TNode : SyntaxNode
```

Comment:
Might want to add its declaration to Interface so that it's required on the Service and not just C# only member.

Reply:
Yes, I would go a step further and even hide this completely from the refactorings and just make it a protected member on `RefactoringHelpersService` which is invoked by `TryGet` APIs.